File: Arrays/Separate_Digits _in_ Array/Space_Optimized_Solution.py
class Solution(object):
    def separateDigits(self, nums):
     n=0
     for i in nums:
        temp=i
        if temp==0:
            n=n+1
        while temp!=0:
            n=n+1
            temp=temp//10

     result=[0]*n
     n=n-1
     for i in range(len(nums)-1,-1,-1):
        temp=nums[i]
        if temp==0:
            result[n]=0
            n=n-1
        while temp!=0:
            result[n]=temp%10
            n=n-1
            temp=temp//10
     return result                           
        

File: Arrays/Separate_Digits _in_ Array/Space_Optimized_Solution.java
class Solution {
    public int[] separateDigits(int[] nums) {
        int n=0;
        for(int i:nums){
            int temp=i;
            if(temp==0){
                n++;
            }
            while(temp!=0){
                n++;
                temp/=10;
            }
        }
        int[] result=new int[n];
        for(int i=nums.length-1;i>=0;i--){
            int temp=nums[i];
            if(temp==0){
                result[--n]=temp;
            }
            while(temp!=0){
                result[--n]=temp%10;
                temp/=10;
            }
        }
        return result; 
    }
}
